gpu-simulator/main.cc

```diff
@@ -81,41 +81,41 @@ int main(int argc, const char **argv) {
 
   unsigned i = 0;
   while (i < commandlist.size() || !kernels_info.empty()) {
-    trace_kernel_info_t *kernel_info = NULL;
-    if (commandlist[i].m_type == command_type::cpu_gpu_mem_copy) {
-      size_t addre, Bcount;
-      tracer.parse_memcpy_info(commandlist[i].command_string, addre, Bcount);
-      std::cout << "launching memcpy command : " << commandlist[i].command_string << std::endl;
-      m_gpgpu_sim->perf_memcpy_to_gpu(addre, Bcount);
-      i++;
-      continue;
-    } else if (commandlist[i].m_type == command_type::kernel_launch) {
-      // Read trace header info for window_size number of kernels
-      while (kernels_info.size() < window_size && i < commandlist.size()) {
-        kernel_trace_t* kernel_trace_info = tracer.parse_kernel_info(commandlist[i].command_string);
-        kernel_info = create_kernel_info(kernel_trace_info, m_gpgpu_context, &tconfig, &tracer);
-        kernels_info.push_back(kernel_info);
-        std::cout << "Header info loaded for kernel command : " << commandlist[i].command_string << std::endl;
+    //gulp up as many commands as possible - either cpu_gpu_mem_copy 
```

Comment:
I think you can remove these 3 lines of comments, it's implied by the loop bounds and isn't really accurate since we don't pop commands off the command list so it never empties and kernels_info will never exceed the window_size

Reply:
I refined my wording to reflect what you are bothered about - yes, `commandlist` never empties, instead we read it element by element.
I think this comment is not only an explanation of what the loop bounds mean, but rather what this chunk of code is doing, so why not keep it for the convenience of others provided the description is accurate? 
By the way I don't understand why kernels_info.size never exceeds the window size.

Reply:
The max length of kernelslist is window_size `(kernels_info.size() < window_size)`, so it never exceeds it.

Reply:
Good catch, I should have used "reached the window_size" instead of "exceeded"